cocos/ui/scroll-bar.ts

```diff
@@ -258,25 +260,31 @@ export class ScrollBar extends Component {
         let outOfBoundaryValue = 0;
         let contentPosition = 0;
         let handleNodeMeasure = 0;
+        const outOfContentPosition = _tempVec2;
+        outOfContentPosition.set(0, 0);
 
         if (this._direction === Direction.HORIZONTAL) {
             contentMeasure = contentSize.width;
             scrollViewMeasure = scrollViewSize.width;
             handleNodeMeasure = barSize.width;
             outOfBoundaryValue = outOfBoundary.x;
 
-            contentPosition = -this._convertToScrollViewSpace(content).x;
+            this._convertToScrollViewSpace(outOfContentPosition, content);
+            contentPosition = -outOfContentPosition.x;
         } else if (this._direction === Direction.VERTICAL) {
             contentMeasure = contentSize.height;
             scrollViewMeasure = scrollViewSize.height;
             handleNodeMeasure = barSize.height;
             outOfBoundaryValue = outOfBoundary.y;
 
-            contentPosition = -this._convertToScrollViewSpace(content).y;
+            this._convertToScrollViewSpace(outOfContentPosition, content);
+            contentPosition = -outOfContentPosition.y;
         }
 
         const length = this._calculateLength(contentMeasure, scrollViewMeasure, handleNodeMeasure, outOfBoundaryValue);
-        const position = this._calculatePosition(contentMeasure, scrollViewMeasure, handleNodeMeasure, contentPosition, outOfBoundaryValue, length);
+        const position = _tempVec2;
+        position.set(0, 0);
```

Comment:
You can remove this line

_calculatePosition will set the position's x, y, so it's safe to directly use it here, no matter what original value it holds.